internal/component/database_observability/mysql/collector/query_sample_test.go

```diff
@@ -402,4 +403,50 @@ func TestQuerySampleSQLDriverErrors(t *testing.T) {
 		err = mock.ExpectationsWereMet()
 		require.NoError(t, err)
 	})
+
+	t.Run("result set iteration error", func(t *testing.T) {
+		db, mock, err := sqlmock.New(sqlmock.QueryMatcherOption(sqlmock.QueryMatcherEqual))
+		require.NoError(t, err)
+		defer db.Close()
+
+		lokiClient := loki_fake.NewClient(func() {})
+
+		collector, err := NewQuerySample(QuerySampleArguments{
+			DB:              db,
+			InstanceKey:     "mysql-db",
+			CollectInterval: time.Second,
+			EntryHandler:    lokiClient,
+			Logger:          log.NewLogfmtLogger(os.Stderr),
+		})
+		require.NoError(t, err)
+		require.NotNil(t, collector)
+
+		mock.ExpectQuery(selectQuerySamples).WithoutArgs().RowsWillBeClosed().
+			WillReturnRows(
+				sqlmock.NewRows([]string{
+					"digest",
+					"query_sample_text",
+					"query_sample_seen",
+					"query_sample_timer_wait",
+				}).AddRow(
+					"abc123",
+					"SELECT 1",
+					"2024-01-01",
+					"1000",
+				).RowError(0, fmt.Errorf("rs error")),
+			)
+
+		err = collector.Start(context.Background())
+		require.NoError(t, err)
+
+		require.Eventually(t, func() bool {
+			return collector.Stopped()
+		}, 5*time.Second, 100*time.Millisecond)
+
+		collector.Stop()
+		lokiClient.Stop()
+
+		err = mock.ExpectationsWereMet()
+		require.NoError(t, err)
```

Comment:
Should we assert the error log has been sent to `lokiEntries := lokiClient.Received()`

Reply:
The error logs are just spit out by Alloy, not sent to the loki client. But this made me realise I could have added another assertion in one of the tests (https://github.com/grafana/alloy/pull/2467/commits/6cbd523a89195b9d67f6917ba15f708bee8e248b).

Reply:
Okay makes sense. I think it would be good to assert that errors are also being logged correctly, but it's out of scope for this change.